### PR TITLE
Feature/state refactor

### PR DESCRIPTION
**Pull request by sdstolworthy**

### CODE REVIEW

**Comment by jacobsee** [comment]
I _think_ it looks good but this is huge... would like to hear @cognizen's thoughts as well.

**Comment by sdstolworthy** [comment]
@jacobsee yeah it's huge :/ I was struggling to find a way to pare it down. Thoughts on how to go about that?

File: src/context/engagement_context/engagement_hook.tsx [hunk 1]
```diff
@@ -0,0 +1,234 @@
+import {
```
Comment (sdstolworthy) [inline src/context/engagement_context/engagement_hook.tsx:1]: The logic from `engagement_context` was moved to this hook. There is not significant net new code in this file, just a consolidation of the logic from `engagement_context` and `engagement_form_context`.

File: src/routes/admin/index.tsx [hunk 2]
```diff
@@ -1,12 +1,12 @@
-import React, {useContext, useEffect} from 'react';
+import React, { useContext, useEffect } from 'react';
```
Comment (sdstolworthy) [inline src/routes/admin/index.tsx:1]: Changes in this file are just from the autoformatter

**Pull request merged**